azure/azure.go

```diff
@@ -388,6 +398,178 @@ func (p azureProvider) GetDatabases() ([]*types.Database, error) {
 	return nil, errors.New("[AZURE] Get databases is not supported")
 }
 
+func (p azureProvider) GetStorages() ([]*types.Storage, error) {
+	accountListResultIterator, err := p.storageAccountClient.ListComplete(context.Background())
+	if err != nil {
+		log.Errorf("[AZURE] Failed to get storage accounts, err: %s", err)
+		return nil, err
+	}
+	storages := []*types.Storage{}
+	for {
+		if accountListResultIterator.NotDone() {
```

Comment:
If you flip the expression, you should do early break, so the business logic should go to the left and the code became more goish.

Reply:
I agree, it is just painful to see `if !NotDone()` :smile:

Reply:
For me it is the same for `} else { break }` :D but i can agree.

Reply:
I've found how it should be used:
`for x.NotDone() {}`

Reply:
Yepp, awesome, thanks